File: frontend/src/types/report.ts
export interface Report {
  report_id: string
  company_id: string
  content: string
  job_ids: string[]
  preferences: {
    interest: string
    additional?: string
  }
  created_at: string
}

export interface GenerateRequest {
  company_id: string
  preferences: {
    interest: string
    additional?: string
  }
}

File: backend/app/schemas/report.py
from pydantic import BaseModel


class Preferences(BaseModel):
    interest: str
    additional: str = ""


class GenerateRequest(BaseModel):
    company_id: str
    preferences: Preferences


class ReportOut(BaseModel):
    report_id: str
    company_id: str
    content: str
    job_ids: list[str]
    preferences: Preferences
    created_at: str
